pkg/controller/migplan/validation.go

```diff
@@ -654,3 +666,60 @@ func containsAccessMode(modeList []kapi.PersistentVolumeAccessMode, accessMode k
 	}
 	return false
 }
+
+// Validate the pods, all should be healthy befor migration
+func (r ReconcileMigPlan) validatePods(plan *migapi.MigPlan) error {
+	if plan.Status.HasAnyCondition(Suspended) {
+		plan.Status.StageCondition(SourcePodsNotHealthy)
+		return nil
+	}
+
+	cluster, err := plan.GetSourceCluster(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	if cluster == nil || !cluster.Status.IsReady() {
+		return nil
+	}
+
+	client, err := cluster.GetClient(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	for _, ns := range plan.Spec.Namespaces {
+		options := k8sclient.InNamespace(ns)
+		podList := kapi.PodList{}
+		err := client.List(context.TODO(), options, &podList)
+		if err != nil {
+			log.Trace(err)
+			return err
+		}
+
+		for _, pod := range podList.Items {
+			if pod.Status.Phase == kapi.PodRunning || pod.Status.Phase == kapi.PodSucceeded {
+				continue
+			}
+
+			condition := plan.Status.FindCondition(SourcePodsNotHealthy)
+			if condition == nil {
+				condition = &migapi.Condition{
+					Type:     SourcePodsNotHealthy,
+					Status:   True,
+					Reason:   NotHealthy,
+					Category: Warn,
+					Message:  "Pods not healthy on the source cluster [].",
```

Comment:
Is the message supposed to have `[]` at the end? Or was the intention to put the cluster name?

Reply:
This part of the condition will be filled with content of the `Items` before being displayed.

Reply:
Ah thank you I didn't realize that's how it worked.

Reply:
Hey Alex, I'd asked Danil to fill you in on the _high level design_ intended to be shared between the (pre) migration validation and the (post) migration verification.  The 1st aspect was for both the migration and plan controllers to leverage the _health_ package which I see you have incorporated.  The 2nd, was to add the `UnhealthyResources` in the MigPlan spec to list the unhealthy namespace instead of listing them in the condition.  The thinking here is that unlike other conditions the list could be very long and it would have symmetry with how similar information is included in the migration.  Please use the `UnhealthyResources` etc in `verify.go`.